src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_signverify.c

```diff
@@ -235,59 +92,190 @@ static int32_t ConfigureSignVerifyTransform(SecTransformRef xform,
         return 0;
     }
 
-    if (includeHashAlgorithm)
-    {
-        CFStringRef cfHashName = NULL;
-        int32_t hashSize = 0;
+    return 1;
+}
+#endif
 
+// Legacy algorithm identifiers
+extern const SecKeyAlgorithm kSecKeyAlgorithmRSASignatureDigestPKCS1v15MD5; // = CFSTR("algid:sign:RSA:digest-PKCS1v15:MD5");
```

Comment:
So, this is a bit tricky. Apple considers PKCS#1 MD5 signatures deprecated but they are still available in the API but not exposed in public headers. There are several possible ways to deal with it:
- Use the private `kSecKeyAlgorithmRSASignatureDigestPKCS1v15MD5` exported string. Works both on iOS and macOS.
- Declare the `kSecKeyAlgorithmRSASignatureDigestPKCS1v15MD5` string with the same contents as the string in the Security framework. Works both on iOS and macOS, avoids using private API.
- Use old-style security transforms for this case just like with DSA. Works only on macOS, avoids using private API.
- Use raw RSA transforms and implement the padding using managed code. Works on both iOS and macOS. There is pre-existing support for similar cases so should not be much more code but may bump the size a little for mobile workloads in future.

Reply:
> Use old-style security transforms for this case just like with DSA. Works only on macOS, avoids using private API.

My two cents: this is probably the option I would have gone with. I don't think RSA PKCS15 with MD5 has a lot of demand on iOS. Using private functionality seems unwise. Even though it is "just a string", the entire functionality could disappear and they would not be breaking an API.

Reply:
> Even though it is "just a string", the entire functionality could disappear and they would not be breaking an API.

That's why the current "extern" version is a bit problematic to me. Specifying the string value is probably slightly better in this regard. If Apple removes the support for it the app will still run but it will throw exceptions from the managed API. It's unlikely that Apple would change the string without dropping the functionality altogether.

Reply:
Either declaring the string locally (and testing what happens for a currently unknown input so we can try to have a handler for it if/when it breaks) or making that one combination fall back to SecurityTransforms is good with me.

The private API latch feels wrong, and might fail an appstore ingestion test.

Reply:
I changed it to a string. I've previously locally tested that an incorrect string results in `CryptographicException` being thrown.